hoplite-core/src/main/kotlin/com/sksamuel/hoplite/SystemPropertiesPropertySource.kt

```diff
@@ -10,14 +10,31 @@ import java.util.Properties
  * In other words, if a System property is defined 'config.override.user.name=sam' then
  * the property 'user.name=sam' is made available.
  */
-object SystemPropertiesPropertySource : PropertySource {
-  private const val prefix = "config.override."
+open class SystemPropertiesPropertySource(
+  private val systemPropertiesMap: () -> Map<String, String> = { System.getProperties().toStringMap() }
+) : PropertySource {
   override fun node(context: PropertySourceContext): ConfigResult<Node> {
     val props = Properties()
-    System.getProperties()
-      .stringPropertyNames()
-      .filter { it.startsWith(prefix) }
-      .forEach { props[it.removePrefix(prefix)] = System.getProperty(it) }
+    systemPropertiesMap().let { systemPropertiesMap ->
+      systemPropertiesMap.keys
+        .filter { it.startsWith(prefix) }
+        .forEach { props[it.removePrefix(prefix)] = systemPropertiesMap[it] }
+    }
     return if (props.isEmpty) Undefined.valid() else props.toNode("sysprops").valid()
   }
+
+  companion object : SystemPropertiesPropertySource() {
+    private const val prefix = "config.override."
+
+    // for backwards compatibility with Java from when SystemPropertiesPropertySource was an `object`
+    @Suppress("unused")
+    @JvmField
+    val INSTANCE: SystemPropertiesPropertySource = this
```

Comment:
Not sure how much you care about backwards compat with Java consumers of the lib. 
Should we add a `.java` test to verify that it works unchanged?

Reply:
It's a kotlin focused library, it works on data classes that don't exist in java anyway.